# HG changeset patch
# User sbr <sebastien@brissaud.name>
# Date 1375248441 -7200
#      Wed Jul 31 07:27:21 2013 +0200
# Node ID c747935615b5daebd80391cf2c1618c7f6bf9b31
# Parent  15e6c83921e55672a946d3e50f00902d56183ae5
-Fix (r21448): The base horizontal offset of NWidgetMatrix children should be shifted by post padding instead of inter padding when drawing in RTL mode.

diff --git a/src/widget.cpp b/src/widget.cpp
--- a/src/widget.cpp
+++ b/src/widget.cpp
@@ -1665,7 +1665,7 @@ void NWidgetMatrix::GetScrollOffsets(int
 		if (this->sb->IsVertical()) {
 			start_y = this->sb->GetPosition() / this->widget_h;
 			base_offs_y = -this->sb->GetPosition() + start_y * this->widget_h;
-			if (_current_text_dir == TD_RTL) base_offs_x = this->pip_pre + this->widget_w * (this->widgets_x - 1) - this->pip_inter;
+			if (_current_text_dir == TD_RTL) base_offs_x = this->pip_pre + this->widget_w * (this->widgets_x - 1) - this->pip_post;
 		} else {
 			start_x = this->sb->GetPosition() / this->widget_w;
 			if (_current_text_dir == TD_RTL) {
